### PR TITLE
compare menus bug fixes

### PR DESCRIPTION
**Pull request by oliviaxjiang**
addressed following bugs in the compare menus bug bash
- selecting one of North/West/Central filter now automatically de-selected the others (if any other campus filters are selected)
- bottom sheet filters lazyrow is no longer cut off due to padding
- added padding between title and filters
- transition from EateryDetailsPage to CompareMenusBottomSheet now automatically selects that eatery and displays it at the top

https://github.com/user-attachments/assets/14ea21ad-319d-4b0d-8a80-d29337ccaa91

(^idk why the eatery images aren't displays in the demo above, might be a internet issue on my end)

bugs that I wasn't able to address
- making horizontal pager less choppy; wasn't able to address because I didn't find a valid solution to fix this issue

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt [hunk 1]
```diff
@@ -63,16 +65,24 @@ class CompareMenusBotViewModel @Inject constructor(
...
+                                        filter.passesFilter(eatery, favorites, selected)
+                                    }
+                                }))
+                            } else {
+                                eateriesApiResponse.data.filter { eatery ->
+                                    filters.all { filter ->
+                                        filter.passesFilter(eatery, favorites, selected)
+                                    }
                                 }
                             },
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt:80]: This code's a bit strange to me. I saw that you called:

```
selectedEateriesFlow.update {
    it + eatery
}
```

below upon initializing first eatery.

Why then do we need to do this `firstEatery` stuff? does it not work without it?

If all this code is doing is basically just putting the first eatery up top, we can likely make it simpler with:

`listOfNotNull(firstEatery)` and `eatery.name != firstEatery?.name`

I'd prefer not having this really fat if else block in the logic here so these just make the if/else logic sort of inlined with null operators.

Also, if REALLY all this is doing is just moving the first eatery up to the top, you can almost definitely do an EVEN simpler approach with `sortedWith` with a comparator that just says that the first eatery should come before any other eatery.

**Review by thisjustin123 — CHANGES_REQUESTED** [review]
One (kind of important) code quality comment--plz take a look! Otherwise all the bugs are fixed very well :-)

**Comment by thisjustin123** [comment]
Oh also, I had to change some gradle things to get the app running. Does it still work on your end? If not, feel free to revert that commit.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt (hunk 1 above)
Comment (oliviaxjiang) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt:80]: thank you, just fixed! I thought about doing sortedWith but the approach i thought of (defining a custom comparator that needs to check firstEatery then sort by alphabetically then by if its open) seems to be more work. Lmk if it still looks lengthy

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt [hunk 2]
```diff
@@ -65,24 +65,13 @@ class CompareMenusBotViewModel @Inject constructor(
...
+                                filters.all { filter ->
+                                    filter.passesFilter(eatery, favorites, selected)
                                 }
                             },
-                            allEateries = if (firstEatery != null) {
-                                listOf(firstEatery!!) + eateriesApiResponse.data.filter { it.name != firstEatery!!.name }
-                            } else {
-                                eateriesApiResponse.data
-                            },
+                            allEateries = listOfNotNull(firstEatery) + eateriesApiResponse.data.filter { it.name != firstEatery?.name },
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt:74]: Noice, works like a charm!

**Review by thisjustin123 — APPROVED** [review]
Good stuff! Can you make sure it runs fine on ur machine before merging? I added a commit for gradle stuff lol

**Comment by oliviaxjiang** [comment]
> Good stuff! Can you make sure it runs fine on ur machine before merging? I added a commit for gradle stuff lol

Yup! the emulator runs smoothly with new gradle stuff

**Pull request merged**